presto-native-execution/presto_cpp/main/TaskResource.cpp

```diff
@@ -277,6 +284,82 @@ proxygen::RequestHandler* TaskResource::createOrUpdateTask(
       });
 }
 
+proxygen::RequestHandler* TaskResource::createOrUpdateBatchTask(
+    proxygen::HTTPMessage* message,
+    const std::vector<std::string>& pathMatch) {
+  return createOrUpdateTaskImpl(
+      message,
+      pathMatch,
+      [this](
+          const protocol::TaskId& taskId,
+          const std::string& updateJson,
+          protocol::TaskUpdateRequest& taskUpdateRequest,
+          velox::core::PlanFragment& planFragment) {
+        protocol::BatchTaskUpdateRequest batchTaskUpdateRequest =
+            json::parse(updateJson);
+        std::shared_ptr<protocol::String> serializedShuffleWriteInfo =
+            batchTaskUpdateRequest.shuffleWriteInfo;
+        std::shared_ptr<operators::ShuffleInterface> shuffleInterface;
+        if (serializedShuffleWriteInfo) {
+          auto shuffleName = SystemConfig::instance()->shuffleName();
+          VELOX_USER_CHECK(
+              !shuffleName.empty(),
+              "Shuffle name not provided from 'shuffle.name' property in "
+              "config.properties");
+          auto& shuffleFactories = operators::ShuffleInterface::factories();
+          auto shuffleFactoryItr = shuffleFactories.find(shuffleName);
+          VELOX_USER_CHECK(
+              shuffleFactoryItr != shuffleFactories.end(),
+              "Shuffle interface with shuffle name '",
+              shuffleName,
+              "' is not registered");
+          auto decodedSerializedShuffleWriteInfo =
+              velox::encoding::Base64::decode(*serializedShuffleWriteInfo);
+          // For shuffle write we will leave memory pool assignment to
+          // operator, passing nullptr here.
+          shuffleInterface = shuffleFactoryItr->second(
```

Comment:
Shall we let operator to create the shuffle interface? We can just store the factory func in the plan node. Otherwise, different operators will share the same shuffle interface instance?